lib/pbench/server/api/resources/__init__.py

```diff
@@ -229,31 +227,44 @@ def convert_username(value: Union[str, None], _) -> Union[str, None]:
     Validate that the user object referenced by the username string exists, and
     return the internal representation of that user.
 
-    The internal representation is the user row ID as a string. If the external
-    value is None, (which means the API's user parameter is nullable), we
-    return None to indicate that instead of attempting to query the User DB.
+    We do not want an unauthenticated client to be able to distinguish between
+    invalid user" (ConversionError here) and "valid user I can't access" (some
+    sort of permission error later). Checking for a valid authentication token
+    here allows rejecting any "user" parameter reference by an unauthenticated
+    user.
+
+    An authenticated user *can* determine whether a username is valid, and will
+    be able to ask for another user's public data with {"user": "name",
+    "access": "public"} (or simply "{"user": "name"}, if the authenticated user
+    doesn't have rights to "name"'s private data). An authenticated user will
+    be given a validation error (BAD_REQUEST/400) if the specified username
+    isn't valid and a permission error (FORBIDDEN) if asking for another user's
+    private data (without ADMIN role).
```

Comment:
That's an interesting question. My interpretation has always been that `NOT_FOUND` means that a resource was properly identified but isn't there. So for `/api/v1/user/<name>` I'd think `NOT_FOUND` would be appropriate; that's a lot less clear when the username is just one of several parameters identifying a resource. We weren't able to even *look* for the resource, because one of your parameters was "unhelpful". That seems more like a `BAD_REQUEST` than a `NOT_FOUND` to me.

Reply:
At line 241, you have `BAD_REQUEST/400` while at line 242 you have just `FORBIDDEN` (no numerical equivalent).  It would probably be good to settle on whether to include the literal or not....